### DISCUSSION TITLE
shodan struggles with multiple queries

### DISCUSSION THREAD

**Discussion by bongobongoland** [projectdiscovery/uncover#207] [Q&A] [answered]
In the queries file for shodan I have about 100 similar requests. I can cram them all into a single query and shodan works (slowly) just fine but if brake them up into shorter queries, the module struggles:

![image](https://user-images.githubusercontent.com/129246896/228965944-272d9ec5-0350-43b2-a904-f59666a981b3.png)
As you can see, shodan returns 503 and 429.

```
# number of times to retry a failed request
retry: 10
```

^^^
I have edited the config file to try to address the issue but the edit seems to be irrelevant in this particular case. Basically, in order to extract all results I have to pack multiple requests into a single one.

**Comment by ehsandeep** [comment]
@bongobongoland Can you retry with the latest release? new changes were made to ensure to have a default rate limit for all sources. 

https://github.com/projectdiscovery/uncover/blob/a3800a09363e86552b82f98807bcd8faf9d38208/sources/session.go#L16-L19

**Comment by ehsandeep** [comment]
also looks it was an issue because of - https://github.com/projectdiscovery/uncover/issues/188

**Comment by bongobongoland** [comment, marked as answer]
I confirm. The issue has been fixed. thanks!
